libcxx/include/__ranges/concepts.h

```diff
@@ -61,6 +62,13 @@ concept borrowed_range =
 template <range _Rp>
 using sentinel_t = decltype(ranges::end(std::declval<_Rp&>()));
 
+#  if _LIBCPP_STD_VER >= 23
+template <range _Rp>
+using const_iterator_t = const_iterator<iterator_t<_Rp>>;
+template <range _Rp>
+using const_sentinel_t = const_sentinel<sentinel_t<_Rp>>;
```

Comment:
Should we implement [LWG3946](https://cplusplus.github.io/LWG/issue3946) together? (Or is it intended to leave it to be done in another PR?)
https://github.com/llvm/llvm-project/blob/445023f1739d3764f1dae6e6b7b0602328198a1e/libcxx/docs/Status/Cxx2cIssues.csv#L28